openssh/client/known_host.sls

```diff
@@ -13,6 +13,7 @@ include:
   - user: {{ user_name }}
   - name: {{ host.name }}
   - enc: {{ host.get('type', 'ecdsa') }}
+  - fingerprint_hash_type: {{ host.get('fingerprint_hash_type', 'md5') }}
```

Comment:
I wouldn't use this parameter unless defined. To use what's default in specific salt version, eg.:

```
fingerprint_hash_type
The public key fingerprint hash type that the public key fingerprint was originally hashed with. This defaults to sha256 if not specified.
New in version 2016.11.4.
Changed in version 2017.7.0:: default changed from md5 to sha256
```

Reply:
How I wrote: this options wasn't in old version of your openssh formula, so all reclass models are with md5 (old default) and without it you must to rewrite/fix all your models, because isn't compatible with new version of salt. I think will be better add backward comptibility plus options for change to new hash type.

Reply:
And what it does if this option is not specified at all? Resp. what's actual problem with updating to newer salt? This way it will by default use old hash type even on new deployments :-/

Reply:
without it we have a problem with state which using rsync (backupninja, barman) during the migration from salt 2016 to 2017. Of course , one way is to change hash to sha256 and still use default option, but these models won't be compatible with old salt, because old salt have different hash by default => you need two models for the same stack, one for salt 2016 and second for 2017 (we have more salt master in one time in our environment).